eodag/plugins/search/build_search_result.py

```diff
@@ -960,6 +957,16 @@ def format_as_provider_keyword(
         }
         return format_query_params(product_type, self.config, available_properties)
 
+    @functools.lru_cache()
+    def fetch_data(self, url: str) -> Any:
+        """
+        Fetches from a provider elements like constraints or forms by using the cache.
+
+        :param url: url from which the constraints can be fetched
+        :returns: cache containing the json file content fetched from the provider
+        """
+        return self._fetch_data(url)
+
```

Comment:
I don't see the point of having both `fetch_data()` and `_fetch_data()` here. What is the point of a separate method ? why not put the decorator directly on `_fetch_data()`? 

The reason we want to avoid applying lru cache to a class instance is explained here:

- https://stackoverflow.com/questions/33672412/python-functools-lru-cache-with-instance-methods-release-object
- https://rednafi.com/python/lru_cache_on_methods/

Hence the  `self.fetch_data = functools.lru_cache()(self._fetch_data)`.
What could be done to keep the function  with `def` is to remove the fetch_data() from the class or use a static method.